zk_test.go

```diff
@@ -68,87 +68,153 @@ func TestIntegration_StateChanges(t *testing.T) {
 
 func TestIntegration_Create(t *testing.T) {
 	ts, err := StartTestCluster(t, 1, nil, logWriter{t: t, p: "[ZKERR] "})
-	if err != nil {
-		t.Fatal(err)
-	}
+	requireNoErrorf(t, err)
 	defer ts.Stop()
+
 	zk, _, err := ts.ConnectAll()
-	if err != nil {
-		t.Fatalf("Connect returned error: %+v", err)
-	}
+	requireNoErrorf(t, err, "ConnectAll()")
 	defer zk.Close()
 
-	path := "/gozk-test"
+	tests := []struct {
+		name          string
+		createFlags   int32
+		specifiedPath string
+		wantErr       string
+	}{
+		{
+			name:        "valid create persistant",
+			createFlags: FlagPersistent,
+		},
+		{
+			name:        "valid container from Create",
+			createFlags: FlagContainer,
+			// NOTE for v2: we dont need CreateContainer method.
+		},
+		{
+			name:          "invalid path",
+			specifiedPath: "not/valid",
+			wantErr:       "zk: invalid path",
+		},
+		{
+			name:        "invalid create ttl",
+			createFlags: FlagTTL,
+			wantErr:     "zk: invalid flags specified",
+		},
+
+		{
+			name:        "invalid flag for create mode",
+			createFlags: 999,
+			wantErr:     "invalid flag value: [999]",
+		},
+	}
+
+	const testPath = "/ttl_znode_tests"
+	// create sub node to create per test in avoiding using the root path.
+	_, err = zk.Create(testPath, nil /* data */, FlagPersistent, WorldACL(PermAll))
+	requireNoErrorf(t, err)
 
-	if err := zk.Delete(path, -1); err != nil && err != ErrNoNode {
-		t.Fatalf("Delete returned error: %+v", err)
-	}
-	if p, err := zk.Create(path, []byte{1, 2, 3, 4}, 0, WorldACL(PermAll)); err != nil {
-		t.Fatalf("Create returned error: %+v", err)
-	} else if p != path {
-		t.Fatalf("Create returned different path '%s' != '%s'", p, path)
-	}
-	if data, stat, err := zk.Get(path); err != nil {
-		t.Fatalf("Get returned error: %+v", err)
-	} else if stat == nil {
-		t.Fatal("Get returned nil stat")
-	} else if len(data) < 4 {
-		t.Fatal("Get returned wrong size data")
+	for idx, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			path := filepath.Join(testPath, fmt.Sprint(idx))
```

Comment:
nit/optional: prefer each subtest specifying a path instead of computing it sometimes, less logic in tests